src/Tizen.Core/Tizen.Core/TCoreTask.cs

```diff
@@ -0,0 +1,797 @@
+        /// if (coreTask == null)
+        /// {
+        ///   coreTask = TCoreTask.Spawn("EventTask");
+        /// }
+        /// 
+        /// int sourceId = coreTask.AddEvent(coreEvent);
+        /// 
+        /// </code>
+        /// </example>
+        /// <since_tizen> 12 </since_tizen>
+        public int AddEvent(TCoreEvent coreEvent)
+        {
+            if (coreEvent == null)
+            {
+                throw new ArgumentNullException(nameof(coreEvent));
+            }
+
+            if (coreEvent.Handle == IntPtr.Zero)
+            {
+                throw new ArgumentException("The event is already added");
+            }
+
+            int id;
+            lock (_idLock)
+            {
+                id = _id++;
+            }
+            var eventSource = new EventSource(id, IntPtr.Zero, coreEvent);
+            _eventMap[id] = eventSource;
+            lock (eventSource)
+            {
+                Interop.LibTizenCore.ErrorCode error = Interop.LibTizenCore.TizenCore.AddEvent(_handle, coreEvent.Handle, out IntPtr handle);
+                if (error != Interop.LibTizenCore.ErrorCode.None)
+                {
+                    _eventMap.TryRemove(id, out var _);
+                    TCoreErrorFactory.CheckAndThrownException(error, "Failed to add an event to the task");
+                }
+
+                eventSource.Handle = handle;
+                coreEvent.Handle = IntPtr.Zero;
+            }
+            return id;
+        }
+
+        /// <summary>
+        /// Removes the registered event from the main loop of the TCoreTask.
+        /// </summary>
+        /// <param name="id">The registered source ID.</param>
+        /// <example>
+        /// <code>
+        /// 
+        /// var coreEvent = new TCoreEvent();
+        /// coreEvent.EventReceived += (sender, args) => {
+        ///   Console.WriteLine("OnEventReceived. ID = {}, Message = {}", args.Id, (string)args.Data);
+        /// };
+        /// 
+        /// var coreTask = TCoreTask.Find("EventTask");
+        /// if (coreTask == null)
+        /// {
+        ///   coreTask = TCoreTask.Spawn("EventTask");
+        /// }
+        /// 
+        /// int sourceId = coreTask.AddEvent(coreEvent);
+        /// ...
+        /// coreTask.RemoveEvent(sourceId);
+        /// 
+        /// </code>
+        /// </example>
+        /// <since_tizen> 12 </since_tizen>
+        public void RemoveEvent(int id)
+        {
+            if (_eventMap.TryRemove(id, out var eventSource))
+            {
+                lock (eventSource)
+                {
+                    if (eventSource.Handle != IntPtr.Zero)
+                    {
+                        Interop.LibTizenCore.TizenCore.RemoveSource(_handle, eventSource.Handle);
+                        eventSource.Handle = IntPtr.Zero;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Emits the event object to a main loop of the TCoreTask.
```

Comment:
how does this relate to Event.Emit() ?

Reply:
It's similar to Event.Emit(), but EmitAllEvent() sends the event object to every event handler of the task while Event.Emit() sends the event object only to the target event's event handler.